### DISCUSSION TITLE
Storage serialize/deserialize and errors

### DISCUSSION THREAD

**Discussion by stephanedupont** [pmndrs/zustand#805] [Q&A] [answered]
Hello,

I lost a few hours because I had a cyclic import problem resulting in an undefined function called in my Zustand storage deserialize function.

No error was thrown, so it took some time to understand and diagnose it.

Is there a silent try/catch happening in Zustand when deserializing the store?

How can we check for errors happening when hydrating the store?

Thank you,

**Comment by AnatoleLucet** [comment, marked as answer]
Hi. If you do not handle errors in [`onRehydrateStorage`](https://github.com/pmndrs/zustand/wiki/Persisting-the-store's-data#onrehydratestorage) they will be dismissed.
Though, it might be a good idea to log the error in case `onRehydrateStorage` is not provided.

**Comment by stephanedupont** [comment]
Oh, I was not aware of this function, sorry, I should have read the doc more carefully!

Thank you.
